internal/output/output_result.go

```diff
@@ -0,0 +1,538 @@
+package output
+
+import (
+	"cmp"
+	"encoding/json"
+	"io"
+	"slices"
+	"strings"
+
+	"github.com/google/osv-scanner/internal/identifiers"
+	"github.com/google/osv-scanner/internal/semantic"
+	"github.com/google/osv-scanner/internal/utility/severity"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+// OutputResult represents the vulnerability scanning results for output report.
+type OutputResult struct { //nolint:revive
+	Ecosystems []OutputEcosystemResult
+	// Container scanning related
+	IsContainerScanning bool
+	AllLayers           []OutputLayerInfo
+	VulnTypeCount       OutputVulnTypeCount
+	VulnCount           OutputVulnCount
+}
+
+// OutputEcosystemResult represents the vulnerability scanning results for an ecosystem.
+type OutputEcosystemResult struct { //nolint:revive
+	Name    string
+	Sources []OutputSourceResult
+	IsOS    bool
+}
+
+// OutputSourceResult represents the vulnerability scanning results for a source file.
+type OutputSourceResult struct { //nolint:revive
+	Name             string
+	Ecosystem        string
+	PackageTypeCount OutputCallAnalysisCount
+	Packages         []OutputPackageResult
+	VulnCount        OutputVulnCount
+}
+
+// OutputPackageResult represents the vulnerability scanning results for a package.
+type OutputPackageResult struct { //nolint:revive
+	Name             string
+	InstalledVersion string
+	FixedVersion     string
+	CalledVulns      []OutputVulnResult
+	UncalledVulns    []OutputVulnResult
+	LayerDetail      OutputPackageLayerDetail
+	VulnCount        OutputVulnCount
+}
+
+// OutputVulnResult represents a single vulnerability.
+type OutputVulnResult struct { //nolint:revive
+	ID             string
+	GroupIDs       []string
+	Aliases        []string
+	FixedVersion   string
+	SeverityRating string
+	SeverityScore  string
+}
+
+// OutputPackageLayerDetail represents detailed layer tracing information about a package.
+type OutputPackageLayerDetail struct { //nolint:revive
+	LayerCommand        string
+	LayerCommandTooltip string
```

Comment:
nit: do we need to call this "Tooltip"? It couples and reveals the implementation detail of the actual frontend in code, which we should try to avoid. 

Would it make sense to call this "LayerCommandDetailed" or something similar?